src/parsing/Parse_rule.ml

```diff
@@ -1476,7 +1476,7 @@ let parse_generic_ast ?(error_recovery = false) (file : Fpath.t)
             let loc = Tok.first_loc_of_file !!file in
             yaml_error (Tok.tok_of_loc loc)
               "missing rules entry as top-level key")
-    | _ -> assert false
+    | _ -> (Tok.(tok_of_loc (first_loc_of_file !!file)), [])
```

Comment:
I'm confused by this case. When does this happen? When the file does not contain anything?
Maybe we should pattern match more precisely on it? and assert false for the rest?
I'm scared we silently skip bad content or are tolerant to bad content (at least the assert false was returning
an error for bad stuff).
Or maybe it's ok because anyway the error will be detected later because we have empty rules? Then maybe
we need a comment there to reinsure the reader the issue will be detected at some point.